client/src/components/Auth/Register/__tests__/Register.test.jsx

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { makeMockStore } from "../../../../test/testUtils";
+import { cleanup, fireEvent, wait } from "@testing-library/react";
+import { renderWithRouter } from "../../../../test/routerTestUtils";
+import { Redirect as MockRedirect } from "react-router-dom";
+import API from "../../../../lib/api";
+import Register from "../Register";
+
+const mockResponse = {
+  success: true,
+  token: "Bearer 5cCI6IkpXVCJ9.oxNTkwNTI4NjA3fQ.1Nd3GGE8"
+};
+
+const defaultProps = {
+  auth: {
+    isAuthenticated: false,
+    user: {},
+    loading: false
+  }
+};
+
+jest.mock("react-router-dom", () => {
+  const original = jest.requireActual("react-router-dom");
+  return {
+    ...original,
+    Redirect: jest.fn().mockImplementation(() => null)
+  };
+});
+
+jest.mock("../../../../lib/api", () => ({
+  post: jest.fn(() => Promise.resolve())
+}));
+
+afterEach(cleanup);
```

Comment:
Can this be done globally, like in a setup file?

Reply:
it can. once i merge this and the tea component tests in, i've got some refactoring to do.